src/main/java/net/hypixel/nerdbot/generator/StringColorParser.java

```diff
@@ -9,310 +8,310 @@
-                            specialSubStringIndex = i;
-                            continue;
-                        }
+                    // checking if the command is a color code
+                    MCColor color = (MCColor) findValue(colors, selectedCommand);
+                    if (color != null) {
+                        this.setColor(color);
+                        charIndex = closingIndex + 2;
+                        continue;
+                    }
 
-                        if (i == 99) {
-                            endCharIndex = -1;
-                            break;
-                        }
+                    // checking if the command is a gemstone type
+                    Gemstone gemstone = (Gemstone) findValue(gemstones, selectedCommand);
+                    if (gemstone != null) {
+                        // replacing the selected space with the stat's text
+                        String replacementText = "%%DARK_GRAY%%" + gemstone.getId() + "%%GRAY%%";
+                        description.replace(charIndex, closingIndex + 2, replacementText);
+                        continue;
+                    }
 
-                        if (i + 2 > description.length()) {
-                            endCharIndex = -1;
-                            break;
-                        }
+                    String extraData = "";
+                    // checking if the command has extra special flags
+                    if (selectedCommand.indexOf(':') != -1) {
+                        int specialFlagIndex = selectedCommand.indexOf(':');
+                        extraData = selectedCommand.substring(specialFlagIndex + 1);
+                        selectedCommand = selectedCommand.substring(0, specialFlagIndex);
+                    }
+                    // checking if the command is a stat
+                    Stat stat = (Stat) findValue(stats, selectedCommand);
+                    if (stat != null) {
+                        // replacing the selected space with the stat's text
+                        String replacementText = "%%" + stat.getColor() + "%%" + extraData + stat.getId() + "%%GRAY%%";
+                        description.replace(charIndex, closingIndex + 2, replacementText);
+                        continue;
                     }
 
-                    // If we can't find the end percents, just continue
-                    if (endCharIndex != -1) {
-                        // Move away from color code
+                    // creating an error message showing the available stats, gemstones and color codes available
+                    StringBuilder failedString = new StringBuilder("You used an invalid code `" + selectedCommand + "`. Valid colors:\n");
+                    for (MCColor availableColors : colors) {
+                        failedString.append(availableColors).append(" ");
+                    }
+                    failedString.append("BOLD");
```

Comment:
make sure we append ITALIC here as well

Reply:
Is there a reason we're not adding these values into MCColor? Technically they are color codes

Reply:
they're special cases because we have to do more logic than just a color change, so i think they should stay seperate from the class

Reply:
There isn't much more logic required because of how the parsed text is shared between the MinecraftImage and StringColorParser. So the `MCColor` class could potentially be renamed to `MCCode`.